subprojects/core/src/main/java/org/gradle/caching/internal/BuildCacheStoreBuildOperationType.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal;
+
+import org.gradle.caching.internal.controller.BuildCacheServiceRole;
+import org.gradle.internal.operations.BuildOperationType;
+import org.gradle.internal.scan.UsedByScanPlugin;
+
+/**
+ * A store operation to a build cache.
+ *
+ * A store operation may STORE or NOT-STORE.
```

Comment:
Actually, what is the intention for dealing with a cache service rejecting a store (e.g. HTTP 413 - Entity Too Large)? We don't really have any way currently for a service to signal that it doesn't want to do the store other than just not invoking the writer, but that doesn't work for this case anyway as the service would have to know the size up front.